src/test/function_test/utils/utils.h

```diff
@@ -201,3 +202,15 @@ inline void compare(const T &expect, const U &actual)
         ASSERT_NO_FATAL_FAILURE(compare(it1->second, it2->second, it1->first));
     }
 }
+
+inline void run_cmd(const std::string &cmd, std::string *output = nullptr)
+{
+    std::stringstream ss;
+    int ret = dsn::utils::pipe_execute(cmd.c_str(), ss);
```

Comment:
In my opinion, it's better to reduce the usage of raw pointer. Using std::string instead is a better choice.

Reply:
Agree, we can refactor functions like `pipe_execute ` in another patch, this patch only refactor function tests.